File: csharp/src/Creational/Builder/BostonSoupBuffetBuilder.cs
//BostonSoupBuffetBuilder.java - One of Two Builder Subclasses

namespace Creational.Builder {

using common;

class BostonSoupBuffetBuilder : SoupBuffetBuilder {
    public new void buildClamChowder() {
       soupBuffet.clamChowder = new BostonClamChowder();
    }
    public new void buildFishChowder() {
       soupBuffet.fishChowder = new BostonFishChowder();
    }    
    
    override public void setSoupBuffetName() {
       soupBuffet.soupBuffetName = "Boston Soup Buffet";
    }
}

}

File: go/creational/Builder/BostonSoupBuffetBuilder.go
//BostonSoupBuffetBuilder - One of Two Builder Subclasses

package builder



type BostonSoupBuffetBuilder struct { // implements SoupBuffetBuilder {
    public void buildClamChowder() {
       soupBuffet.clamChowder = new BostonClamChowder()
    }
    public void buildFishChowder() {
       soupBuffet.fishChowder = new BostonFishChowder()
    }    
    
    public void setSoupBuffetName() {
       soupBuffet.soupBuffetName = "Boston Soup Buffet"
    }
}
